### PR TITLE
Query thread log doc

### PR DESCRIPTION
**Pull request by Felixoid**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Documentation

Short description (up to few sentences):

Add documentation for `system.query_thread_log` table

### CODE REVIEW

File: docs/ru/operations/system_tables.md [hunk 1]
```diff
@@ -477,6 +477,76 @@ ClickHouse создаёт таблицу только в том случае, к
...
+
+- `event_date` (Date) — дата события.
+- `event_time` (DateTime) — время события.
+- `query_start_time` (DateTime) — время начала обработки запроса.
+- `query_duration_ms` (UInt64) — длительность обработки запроса.
+- `read_rows` (UInt64) — количество прочитанных строк.
+- `read_bytes` (UInt64) — количество прочитанных байтов.
+- `written_rows` (UInt64) — количество записанных строк для запросов `INSERT`. Для других запросов, значение столбца 0.
+- `written_bytes` (UInt64) — объем записанных данных в байтах для запросов `INSERT`. Для других запросов, значение столбца 0.
+- `memory_usage` (Int64) — Потребление RAM тредом (?).
```
Comment (alexey-milovidov) [inline docs/ru/operations/system_tables.md]: Запросом, не потоком.

File: docs/en/operations/system_tables.md [hunk 2]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `event_date` (Date) — Event date.
+- `event_time` (DateTime) — Event time.
+- `query_start_time` (DateTime) — Start time of query execution.
+- `query_duration_ms` (UInt64) — Duration of query execution.
+- `read_rows` (UInt64) — Number of read rows.
+- `read_bytes` (UInt64) — Number of read bytes.
+- `written_rows` (UInt64) — For `INSERT` queries, the number of written rows. For other queries, the column value is 0.
+- `written_bytes` (UInt64) — For `INSERT` queries, the number of written bytes. For other queries, the column value is 0.
+- `memory_usage` (Int64) — Memory consumption by the whole query.
+- `peak_memory_usage` (Int64) — Maximum memory consumption by the thread.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: For query.

File: docs/en/operations/system_tables.md [hunk 3]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `thread_name` (String) — Name of the thread function.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Not function. Just name of the thread.

File: docs/en/operations/system_tables.md [hunk 4]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `master_thread_number` (UInt32) — Internal ID of initial thread.
+- `master_os_thread_id` (Int32) — OS initial thread ID.
+- `query` (String) — Query string.
+- `is_initial_query` (UInt8) — Query type. Possible values:
+    - 1 — Query was initiated by the client.
+    - 0 — Query was initiated by another query for distributed query execution.
+- `user` (String) — Name of the user who initiated the current query.
+- `query_id` (String) — ID of the query.
+- `address` (FixedString(16)) — IP address the query was initiated from.
+- `port` (UInt16) — The server port that was used to receive the query.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: No, this is the port on client side.

File: docs/en/operations/system_tables.md [hunk 5]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `initial_user` (String) —  Name of the user who ran the parent query (for distributed query execution).
+- `initial_query_id` (String) — ID of the parent query.
+- `initial_address` (FixedString(16)) — IP address that the parent query was launched from.
+- `initial_port` (UInt16) — The server port that was used to receive the parent query from the client.
+- `interface` (UInt8) — Interface that the query was initiated from. Possible values:
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: BTW all these fields are shared with query_log.

File: docs/en/operations/system_tables.md [hunk 6]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+    - 1 — TCP.
+    - 2 — HTTP.
+- `os_user` (String) — User's OS.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: No. It's the user name who has run the clickhouse-client.

File: docs/en/operations/system_tables.md [hunk 7]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `client_hostname` (String) — Server name that the [clickhouse-client](../interfaces/cli.md) is connected to.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Wrong.

File: docs/en/operations/system_tables.md [hunk 8]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `client_name` (String) — The [clickhouse-client](../interfaces/cli.md) name.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Wrong.

Comment (alexey-milovidov) [reply docs/en/operations/system_tables.md]: It is the name of the client, which can be clickhouse-client, clickhouse-benchmark or whatever else.

File: docs/en/operations/system_tables.md [hunk 9]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `client_revision` (UInt32) — Revision of the [clickhouse-client](../interfaces/cli.md).
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Not necessarily of clickhouse-client.

File: docs/en/operations/system_tables.md [hunk 10]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `client_version_major` (UInt32) — Major version of the [clickhouse-client](../interfaces/cli.md).
+- `client_version_minor` (UInt32) — Minor version of the [clickhouse-client](../interfaces/cli.md).
+- `client_version_patch` (UInt32) — Patch component of the [clickhouse-client](../interfaces/cli.md) version.
+- `http_method` (UInt8) — HTTP method that initiated the query. Possible values:
+    - 0 — The query was launched from the TCP interface.
+    - 1 — `GET` method was used.
+    - 2 — `POST` method was used.
+- `http_user_agent` (String) — The `UserAgent` header passed in the HTTP request.
+- `quota_key` (String) — The quota key specified in the [quotas](quotas.md) setting.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Slightly incorrect. This is the user specified arbitrary string that is used as the key for quota.

File: docs/en/operations/system_tables.md [hunk 11]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+- `revision` (UInt32) — ClickHouse revision.
+- `ProfileEvents.Names` (Array(String)) — Counters that measure the following metrics:
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: This is not the complete list.

File: docs/en/operations/system_tables.md [hunk 12]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+    - Time spent on reading and writing to a disk.
+    - Number of network errors.
+    - Time spent on waiting when the network bandwidth is limited.
+- `ProfileEvents.Values` (Array(UInt64)) — Values of metrics that are listed in the&#160;`ProfileEvents.Names` column.
+
+By default, logs are added to the table at intervals of 7.5 seconds. You can set this interval in the [query_thread_log](server_settings/settings.md#server_settings-query-thread-log) server setting (see the `flush_interval_milliseconds` parameter). To flush the logs forcibly from the memory buffer into the table, use the `SYSTEM FLUSH LOGS` query.
+
+When the table is deleted manually, it will be automatically created on the fly. Note that all the previous logs will be deleted.
+
+!!! note
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: A link to TTL.

File: docs/en/operations/system_tables.md [hunk 13]
```diff
@@ -482,6 +482,76 @@ When the table is deleted manually, it will be automatically created on the fly.
...
+
+The table contains information about each query execution thread.
+
+ClickHouse creates this table only if the [query_thread_log](server_settings/settings.md#server_settings-query-thread-log) server parameter is specified. This parameter sets the logging rules, such as the logging interval or the name of the table the queries will be logged in.
+
+To enable query logging, set the [log_query_threads](settings/settings.md#settings-log-query-threads) parameter to 1. For details, see the [Settings](settings/settings.md) section.
+
+Columns:
+
+- `event_date` (Date) — Event date.
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: What kind of event?

File: docs/en/operations/system_tables.md (hunk 2 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: fixed

File: docs/en/operations/system_tables.md (hunk 3 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: fixed

File: docs/en/operations/system_tables.md (hunk 4 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Done

File: docs/en/operations/system_tables.md (hunk 5 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: I've copied them from that part

File: docs/en/operations/system_tables.md (hunk 6 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Done

File: docs/en/operations/system_tables.md (hunk 7 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Fixed

File: docs/en/operations/system_tables.md (hunk 8 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Copied from query_log, fixed in both places

File: docs/en/operations/system_tables.md (hunk 9 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Fixed as well

File: docs/en/operations/system_tables.md (hunk 12 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Is it possible to alter *_log tables with setting TTL? https://clickhouse.yandex/docs/en/operations/table_engines/mergetree/#table_engine-mergetree-ttl

Or is it possible to set it in the config directly?

**Review by Felixoid — COMMENTED** [review]
Fixed most of the points and rebased the branch on master. There is still some question. And I would like to squash the commits before the merge.

File: docs/en/operations/system_tables.md (hunk 11 above)
Comment (Felixoid) [reply docs/en/operations/system_tables.md]: Fixed

**Comment by blinkov** [comment]
@Felixoid at least there are some file conflicts

**Comment by Felixoid** [comment]
@blinkov conflicts are resolved

File: docs/en/operations/system_tables.md (hunk 12 above)
Comment (alexey-milovidov) [reply docs/en/operations/system_tables.md]: It's possible to ALTER to add TTL. It's not possible to specify in config directly.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**